rmf_site_editor/src/lib.rs

```diff
@@ -212,28 +224,24 @@ impl Plugin for SiteEditor {
                 AabbUpdatePlugin,
                 EguiPlugin,
                 KeyboardInputPlugin,
-                MainMenuPlugin,
-                WorkcellEditorPlugin,
                 SitePlugin,
-                InteractionPlugin {
-                    headless: self.headless_export.is_some(),
-                },
-                StandardUiLayout {
-                    headless: self.headless_export.is_some(),
-                },
+                InteractionPlugin::new().headless(self.headless_export.is_some()),
                 AnimationPlugin,
                 OccupancyPlugin,
                 WorkspacePlugin,
-            ))
-            // Note order matters, plugins that edit the menus must be initialized after the UI
-            .add_plugins((
-                ViewMenuPlugin,
                 IssuePlugin,
-                OSMViewPlugin,
-                SiteWireframePlugin,
-                SiteFileMenuPlugin,
             ));
 
+        if self.headless_export.is_none() {
+            app.add_plugins((
+                StandardUiPlugin::default(),
```

Comment:
This stands out from the others and I believe it wouldn't be needed if the struct definition changed to a unit struct:

```
// Before
struct StandardUiPlugin { }
// After
struct StandardUiPlugin;
```

So just pondering, any reason this was done this way (I could guess that in the future we could add fields to the plugin without having to break API if users are forced to call a `default()` constructor) or can we make it consistent with the other plugins?

Reply:
> I could guess that in the future we could add fields to the plugin without having to break API if users are forced to call a default() constructor

Yes, this is exactly the rationale, and this is the same reason I applied this pattern to all the plugins in the `widget` module. Using a unit struct creates an API contract that says "I will never add any fields to this struct". This is too rigid of a contract for us to ensure if we want to work towards providing a stable public API.

> can we make it consistent with the other plugins?

My intention is to actually make all the other plugins consistent with this pattern. I could do that in this PR, but I figured I would save it for a follow-up PR to avoid too much noise in this one.